include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -527,6 +527,29 @@ typedef struct OrtApiBase OrtApiBase;
 */
 ORT_EXPORT const OrtApiBase* ORT_API_CALL OrtGetApiBase(void) NO_EXCEPTION;
 
+/** \brief Thread work loop function
+*
+* Onnxruntime will provide the working loop on custom thread creation
+* Argument is an onnxruntime built-in type which will be provided when thread pool calls CustomCreateThreadFn
+*/
+typedef void (*OrtThreadWorkerFn)(void* worker_fn_param);
+
+#define THREAD_HANDLE void* // either a windows handle or a posix pthread_t
+
+/** \brief Thread creation function
```

Comment:
```suggestion
/** \brief Custom thread creation function
```
Also, please document why and when I should be using this API.

Reply:
Done.